yt/fields/field_info_container.py

```diff
@@ -397,4 +397,19 @@ def check_derived_fields(self, fields_to_check = None):
             mylog.debug("Succeeded with %s (needs %s)", field, fd.requested)
         dfl = set(self.ds.derived_field_list).union(deps.keys())
         self.ds.derived_field_list = list(sorted(dfl, key=tupleize))
+        self.set_linear_fields()
```

Comment:
I wasn't sure if this was the best place to call this function.  But it seemed to work.  I just want to assure it gets called when a new dataset is loaded and after all of the fields are defined.